pipeline/matching/generators/limited_schedules.py

```diff
@@ -7,13 +7,78 @@
 
 
 class LimitedSchedulesGenerator(MatchGenerator):
-    def __init__(self, max_available: int = 0):
+    def __init__(self, max_available: int = 1):
         # Maximum number of available slots in a user's schedule for them to be
         # eligible for a limited availability match.
         self.max_available = max_available
         super().__init__(name=GENERATOR_LIMITED_SCHEDULES)
 
     def generate(self, inp: MatchingInput) -> Iterator[Match]:
         # TODO: Implement your generator
-        inp.logger.info("Luke Arrington was here!")
+
+        # get limited availability users
+        limited_availability = []
+        for user in inp.users:  # for users in the list users
+            if (
+                len(user.schedule) <= self.max_available
+            ):  # if the number of the user schedule is less than or equal to the max available which is one
+                limited_availability.append(
+                    user
+                )  # then consider that a user a limited schedule user and add to the list
+        inp.logger.info(limited_availability)
+
+        matches = {}
+        # compare limited users to all users
+        for limited_user in limited_availability:
+            for user in inp.users:
+                if (
+                    limited_user.uid != user.uid
+                ):  # check that we are not looking at the current limited_user
+                    # compare limited user schedule to schedule of user in all users
+                    for l_availability in limited_user.schedule:
+                        for availability in user.schedule:
+                            if (
+                                l_availability == availability
+                            ):  # if the availabilities in limited user schedule and other user schedule are the same, then we have a match!
+                                if (
+                                    limited_user.uid,
+                                    user.uid,
+                                ) in matches:  # add availability to availability metadata
+                                    pass
+                                else:
+                                    # add first match of this set of users to dictionary
+                                    # TODO: add the availability to this first match
+                                    matches[
+                                        {limited_user.uid, user.uid}
+                                    ] = Match(
+                                        users={limited_user.uid, user.uid}
```

Comment:
When you are ready to create the matches, you can add the availabilities the two users have in common like this:

```python
# Put this at the top of your file to import the metadata (or add it to the list of types already being imported)
from pipeline.types import MatchMetadata

Match(
  users={uid1, uid2},
  metadata=MatchMetadata(
    generator=GENERATOR_LIMITED_SCHEDULES,
    availability=availability
  )
)
```

The variable `availability` should be a list of the matching availabilities between the users, from their schedules.